src/rtc/rtc_s1_single_job.py

```diff
@@ -1095,14 +1169,28 @@ def run_single_job(cfg: RunConfig):
         logger.info(f'Processing burst: {burst_id} ({burst_index+1}/'
                     f'{n_bursts})')
 
-        burst_id_file_name = burst_id.upper().replace('_', '-')
-        burst_product_id = \
-            product_id.replace('{burst_id}', burst_id_file_name)
+        geogrid = cfg.geogrids[burst_id]
+        pol_list = list(burst_pol_dict.keys())
+        burst = burst_pol_dict[pol_list[0]]
+ 
+
+
+        # populate burst_product_id
+        pixel_spacing_avg = int((geogrid.spacing_x + geogrid.spacing_y) / 2)
+        burst_product_id = populate_product_id(
+            runconfig_product_id, burst, processing_datetime, product_version,
+            pixel_spacing_avg, is_mosaic=True)
+
+
+
 
```

Comment:
nit pick - please remove blank likes longer than 1~2 lines, unless it is absolutely necessary.